common/common-api/src/main/java/org/ow2/proactive/scripting/SelectionScript.java

```diff
@@ -374,29 +373,21 @@ public boolean compareByteArray(byte[] array1, byte[] array2) {
     }
 
     /**
-     * Util method to get an hashCode of a list of selection script. Can be used to compare two lists of selectionScript
-     * and see if it is the same.
-     * If the list is empty or null, then the return value will be 0, meaning no selection scripts are provided.
-     *
-     * @param selScriptsList a list of selection scripts
-     * @return 0 if the list is null or empty, otherwise, an hashCode of the list content
+     * Get MD5 hash value
+     * 
+     * @return MD5 hash value
+     * @throws NoSuchAlgorithmException
      */
-    public static int hashCodeFromList(List<SelectionScript> selScriptsList) {
-        if (selScriptsList == null || selScriptsList.size() == 0) {
-            return 0;
-        }
-        int toReturn = 0;
-        for (SelectionScript ss : selScriptsList) {
-            toReturn += ss.hashCode();
-        }
-        return toReturn;
+    public byte[] digest() throws NoSuchAlgorithmException {
+        return MessageDigest.getInstance("MD5").digest(getHash());
     }
 
-    /**
-     * String representation
-     */
+    @Override
     public String toString() {
-        return hashCode() + "\n" + script;
+        if (script != null) {
+            return "" + new String(getHash()) + "\n" + script;
+        } else {
+            return "" + new String(getHash()) + "\n" + url;
+        }
```

Comment:
Can be replaced by: 
return script != null ? "" + new String(getHash()) + "\n" + script : "" + new String(getHash()) + "\n" + url;